automations/js/src/utils/octokit.mjs

```diff
@@ -0,0 +1,17 @@
+import { Octokit } from 'octokit'
+
+/** the personal access token for the GitHub API */
+const pat = process.env.ACCESS_TOKEN
+if (!pat) {
+  console.error('GitHub personal access token "ACCESS_TOKEN" is required.')
+  process.exit(1)
+}
+
+/**
+ * Get an authenticated instance of `Octokit` to make GitHub API calls.
+ *
+ * @returns {Octokit} an authenticated instance of `Octokit`
+ */
+export function getOctokit() {
+  return new Octokit({ auth: pat })
+}
```

Comment:
I'll see if using `actions/github-script` can help here. I'm not too sure (especially for the PRs side which runs using `workflow-run` and thus needs context to be provided to it via artifacts) but it will be interesting to explore that.

Reply:
I converted the script to `actions/github-script` to explore and it has had the following impact.

- Changes to surrounding code were reduced, meaning 8 fewer files to review.
- The new files themselves became a bit more complex to be able to be referenced from the action.

Let me know if you think this is a net improvement or if we should go back to aaf807f.

Reply:
To me, it does read as an improvement, and I don't find it harder to reference from the action, but do acknowledge it makes it much harder (maybe just impossible) to run them locally. https://github.com/nektos/act could be a workaround for that, potentially even a better approach because it doesn't rely on manually configuring event payloads, which would be easy to get incorrect. We don't have good ways to test any other part of our CI workflows, so these ones wouldn't be unique.

I think it basically has a net-zero effect and either approach is fine, so totally your preference. My approval of the original one was genuine :slightly_smiling_face:

Reply:
In that case, let's stick with the changes. Considerable brainpower and time had to be invested (as must be evident from the commits) to get to where it is. Thanks for the pointer, it reduced the changed files by a third!